presentation/src/main/java/com/sopt/presentation/mypage/editProfile/EditProfileViewModel.kt

```diff
@@ -37,8 +39,17 @@ class EditProfileViewModel @Inject constructor(
     }
 
     fun navigateToMyPage() {
-        saveNickname(_userInfoState.value.nickname)
-        emitSideEffect(EditProfileSideEffect.NavigateToMyPage)
+        executeInScope {
+            saveUserInfo()
+            withContext(Dispatchers.Main) {
+                emitSideEffect(EditProfileSideEffect.NavigateToMyPage)
+            }
+        }
```

Comment:
P2: 이게 더 가독성이 좋을 것 같아유!! 근데 메인 스레드 navigation 코드를 넣은 이유가 먼가유
```suggestion
        viewModelScope.launch {
            saveUserInfo()
            emitSideEffect(EditProfileSideEffect.NavigateToMyPage)
        }
```

Reply:
저장 작업 완료 후 네비게이션 호출하기 위해 withContext를 사용했습니다!
그냥 emitSideEffect로 화면을 이동하면 Mypage에서 데이터를 불러올 때 최신 데이터가 반영되지 않는 경우가 있더라구요!
호옥시 더 좋은 방법 있으면 알려주세요...